### DISCUSSION TITLE
[TableauOperator] Use params by template_fields

### DISCUSSION THREAD

**Discussion by seunggabi** [apache/airflow#29353] [Ideas] [answered]
I want to use dag params on `TableauOperator`.
So, I suggest this code.

### as-is

```python
task_refresh_workbook_blocking = TableauOperator(
    resource="workbooks",
    method="refresh",
    match_with="id",
    find="MyWorkbook",
    blocking_refresh=True,
    task_id="refresh_tableau_workbook_blocking",
    dag=dag,
)
```

### to-be

```python
task_refresh_workbook_blocking = TableauOperator(
    resource="workbooks",
    method="refresh",
    match_with="{{ params.match_with }}",
    find="{{ params.find }}",
    blocking_refresh=True,
    task_id="refresh_tableau_workbook_blocking",
    dag=dag,
)
```

**Comment by Taragolis** [comment, marked as answer]
Option 1: Quick solution. 
---

**overwrite class attribute** 
```python
from airflow.providers.tableau.operators.tableau import TableauOperator

TableauOperator.template_fields = {"match_with", "find"}
```

**create new operator based on TableauOperator** 
```python
from airflow.providers.tableau.operators.tableau import TableauOperator

class AwesomeTableauOperator(TableauOperator):
    template_fields = {"match_with", "find"}
```

Option 2: Long-term Solution
---

Make other potential users of Tableau Provider happy and make a PR with appropriate changes in TableauOperator.

**Comment by seunggabi** [comment]
@Taragolis thx to answer!

I'll make a pr, your option.1~!

**Reply by Taragolis** [reply]
If it would your first PR to Airflow, i suggest to read [Contributor's Quick Start](https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst) and all relevant docs.

**Comment by seunggabi** [comment]
@Taragolis 

I make first pr. https://github.com/apache/airflow/pull/29360
Thx about your guide.
